### PR TITLE
[LEVWEB-1334] Add group filter

### PR DESCRIPTION
**Pull request by aarongill1**

### CODE REVIEW

File: src/lev-report/ListFilter.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+ import React from 'react';
+import styled from 'styled-components';
+import { InputField } from 'govuk-react';
+
+const StyledListFilter = styled(InputField)`
```
Comment (daniel-ac-martin) [inline src/lev-report/ListFilter.js]: StyledInputField?

File: src/lev-report/ListFilter.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const ListFilter = ({ hint, children, input}) =>
+<StyledListFilter
+    hint={hint}
+    input={{
+        ...input,
+        list: "list"
+    }}
+    >
+    <datalist id="list">
```
Comment (daniel-ac-martin) [inline src/lev-report/ListFilter.js]: `list` could clash quite easily.

Perhaps `group-filter` would be a bit better?

Do we need an id set at all?

File: src/lev-report/ListFilter.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {input.groups.map((item) =>
+          <option value={item.name.replace(/\//g, '')}></option>
```
Comment (daniel-ac-martin) [inline src/lev-report/ListFilter.js]: I think I'd just keep the slashes in place.

File: src/lev-report/ReportFilters.js [hunk 4]
```diff
@@ -10,9 +11,10 @@ const FormDiv = styled.div`
 `;
 const BottomButton = styled('Button')`
   margin-bottom: 2px;
-  position: absolute;
+  display: inline-block;
   bottom: 0px;
```
Comment (daniel-ac-martin) [inline src/lev-report/ReportFilters.js]: Do we still need this line?

File: src/lev-report/ReportFilters.js [hunk 5]
```diff
@@ -36,6 +38,13 @@ const ReportFilters = props =>
     }}>
       Usage to
     </DatePicker>
+    <ListFilter hint="Default: All Departments" input={{
+        name: 'currentGroup',
+        groups: props.groups,
```
Comment (daniel-ac-martin) [inline src/lev-report/ReportFilters.js]: Should `groups` be a part of `input` or should it be its own thing?

I'd think it should be renamed to `options` to be a bit less specific as well.

File: src/lev-report/UsageCounter.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledCounter = styled.div`
```
Comment (daniel-ac-martin) [inline src/lev-report/UsageCounter.js]: I'd call this `StyledDiv` or just `Container` or something. It's not a 'Counter' at this point.

File: src/lev-report/UsageCounter.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  }
+`;
+
+const formatCount = (count) => {
+return count.toString().replace(/\d(?=(\d{3})+$)/g, '$&,')
+}
+
+const UsageCounter = props =>
+    <StyledCounter>
+        <p>{props.countPeriod}</p>
```
Comment (daniel-ac-martin) [inline src/lev-report/UsageCounter.js]: `props.period`?

File: src/lev-report/UsageCounter.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        {props.group ? <p>{props.group}</p> : ''}
```
Comment (daniel-ac-martin) [inline src/lev-report/UsageCounter.js]:
```jsx
{props.group && <p>{props.group}</p>}
```
Should work I think.

File: src/lev-report/UsageCounter.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  p {
+    font-size: 20px;
+    margin-top: 0;
+    margin-bottom: 5px;
+  }
+`;
+
+const formatCount = (count) => {
+return count.toString().replace(/\d(?=(\d{3})+$)/g, '$&,')
+}
```
Comment (daniel-ac-martin) [inline src/lev-report/UsageCounter.js]:
```js
const formatCount = count => count.toString().replace(/\d(?=(\d{3})+$)/g, '$&,');
```

File: src/lev-report/index.js [hunk 10]
```diff
@@ -3,55 +3,80 @@ import LevPage from '../lev-page';
...
+                    <UsageCounter count={this.state.todaySearches[0].count}
+                                  countPeriod="Searches Today" group={this.state.currentGroup} />
+                    <UsageCounter count={this.state.allTime[0].count}
+                                  countPeriod="Total Searches" group={this.state.currentGroup} />
+                </CountDiv>
+            </RelativeParent>
+            <LevUsage dates={this.state.dates} datasets={this.state.datasets}/>
+            <UsageByGroup datasets={this.state.datasets} groups={this.state.groups} totals={this.state.totals}/>
+        </LevPage>;
+    }
```
Comment (daniel-ac-martin) [inline src/lev-report/index.js]: I think your editor has changed the indent from 2 spaces to 4 which makes this hard to review.

File: src/lev-report/ListFilter.js (hunk 1 above)
Comment (aarongill1) [reply src/lev-report/ListFilter.js]: Now a data-list so changed to StyledDataList

File: src/lev-report/ReportFilters.js (hunk 4 above)
Comment (aarongill1) [reply src/lev-report/ReportFilters.js]: No, has been removed

File: src/lev-report/UsageCounter.js (hunk 6 above)
Comment (aarongill1) [reply src/lev-report/UsageCounter.js]: Changed

File: src/lev-report/UsageCounter.js (hunk 8 above)
Comment (aarongill1) [reply src/lev-report/UsageCounter.js]: Has been updated with some conditionals that include text, not just the props.group, so left as is.

File: src/lev-report/index.js (hunk 10 above)
Comment (aarongill1) [reply src/lev-report/index.js]: Done

File: src/lev-report/DateRange.js [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+  import React from 'react';
```
Comment (daniel-ac-martin) [inline src/lev-report/DateRange.js]: I think this extra space should be removed.

File: src/lev-report/ListFilter.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+ import React from 'react';
```
Comment (daniel-ac-martin) [inline src/lev-report/ListFilter.js]: This extra whitespace should also be removed.

File: src/lev-report/UsageCounter.js (hunk 7 above)
Comment (aarongill1) [reply src/lev-report/UsageCounter.js]: countPeriod is no longer a prop, so resolved.

File: src/lev-report/DateRange.js (hunk 11 above)
Comment (aarongill1) [reply src/lev-report/DateRange.js]: Done

File: src/lev-report/ListFilter.js (hunk 12 above)
Comment (aarongill1) [reply src/lev-report/ListFilter.js]: Done

File: src/lev-report/UsageCounter.js (hunk 6 above)
Comment (daniel-ac-martin) [reply src/lev-report/UsageCounter.js]: Have you commited and pushed the change? It doesn't seem to have updated.

Comment (aarongill1) [reply src/lev-report/UsageCounter.js]: No, I'm shortly going to after I fix another issue...

File: src/lev-report/ListFilter.js (hunk 2 above)
Comment (aarongill1) [reply src/lev-report/ListFilter.js]: Fixed. id is needed to associate list with input

File: src/lev-report/ListFilter.js (hunk 3 above)
Comment (aarongill1) [reply src/lev-report/ListFilter.js]: Changed RegEx to remove first / and replace others with ' - '

File: src/lev-report/ReportFilters.js (hunk 5 above)
Comment (aarongill1) [reply src/lev-report/ReportFilters.js]: Refactored to put groups into component props as opposed to input

File: src/lev-report/UsageCounter.js (hunk 9 above)
Comment (aarongill1) [reply src/lev-report/UsageCounter.js]: Done

**Comment by RobinKnipe** [comment]
Marking as `WIP`, until the `lev-report` changes are also ready, and the combination has been tested...

**Pull request merged**